test/files/rules/no-duplicate-variable.test.ts

```diff
@@ -151,4 +151,9 @@ function testDestructuring() {
 
     var [a2, [b2, c2]] = [1, [2, 3]];
     var [{a2, d2}] = [{a2: 1, d2: 4}]; // failure
+
+    function myFunc2([a, b]) {
+        var a; // not a failure; caught by no-shadowed-variable
```

Comment:
in addition: we don't want any overlap in our rules, otherwise errors would be reported twice. `no-duplicate-variable` now checks for duplicates in the same block scope only, rather than across scopes.